src/extensions/menu.extension.ts

```diff
@@ -17,8 +20,32 @@ export class Menu {
     options?: {
       level?: number;
       headline?: string;
+      showHelp?: boolean;
+      helpLabel?: string;
+      backLabel?: string;
+      cancelLabel?: string;
+      byeLabel?: string;
     }
   ) {
+    const cacheIsEmpty = !Object.keys(this.optionsCache).length;
+    if (cacheIsEmpty) {
```

Comment:
You may not want to check here for missing properties, but rather for new ones given by `options`, as if:
```
if (options && options.setCache && (!!options.showHelp || !!options.helpLabel || !!options.backLabel || options.cancelLabel || !!options.byeMessage)) {
```

Reply:
Done!